App/LoginValidate.js

```diff
@@ -0,0 +1,73 @@
+var LOGIN_SUCCESFULLY = 0;
+var INVALID_PASSWORD = 1;
+var INVALID_USERNAME = 2;
+
+var usernameJson = JSON.parse(usersJSON);
+// console.log(usernameJson);
+
+var accountService = AccountService.getInstance();
+
+document.addEventListener("DOMContentLoaded", function(event) {
+
+  for (var i = 0; i < usernameJson.length; i++) {
+    var user = new User(usernameJson[i]);
+    // console.log("adduser");
+    AccountService._instance.addUser(user);
+    // console.log(accountService);
+  }
+  });
+
+
+function logOutUser() {
+  localStorage.setItem("nameUser1", null);
+  window.location.href = "index.html";
+}
+
+
+function validate()
+{
+    //console.log(accountService.usersArray);
+  var username = document.getElementById("username").value;
+  var password = document.getElementById("password").value;
+  AccountService._instance.setUsername(username);
```

Comment:
Please change all refs to AccountService._instance to use getInstance which also ensures that _instance actually exists besides being the public interface